website/projects/map4sci/src/app/app.component.ts

```diff
@@ -1,10 +1,90 @@
-import { Component } from '@angular/core';
+import { Any } from '@angular-ru/common/typings';
+import { Component, OnInit } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { FeatureCollection } from 'geojson';
 
+const EMPTY_FEATURES: FeatureCollection = {
+  type: 'FeatureCollection',
+  features: []
+};
+
+const EMPTY_DATASET = {
+  id: 'empty',
+  boundary: EMPTY_FEATURES,
+  cluster: EMPTY_FEATURES,
+  edges: EMPTY_FEATURES,
+  nodes: EMPTY_FEATURES
+}
+
+interface MapDataset {
+  boundary: FeatureCollection;
+  cluster: FeatureCollection;
+  edges: FeatureCollection;
+  nodes: FeatureCollection;
+  id: string;
+}
+
+interface MapDatasetDirectory {
+  [key: string]: MapDataset;
+  [index: number]: MapDataset;
+}
 
 @Component({
   selector: 'm4s-root',
   templateUrl: './app.component.html',
   styleUrls: ['./app.component.scss']
 })
-export class AppComponent {
+export class AppComponent implements OnInit {
+  constructor(private readonly http: HttpClient) { }
+
+  files: string[] = [
+    'boundary',
+    'cluster',
+    'edges',
+    'nodes'
+  ];
+
+  dataset: Any = { };
+  datasetLookup: Any[] = [];
+  currentDataset: MapDataset = EMPTY_DATASET;
+
+  async ngOnInit(): Promise<void> {
+    this.datasetLookup = await this.getMapData();
+    this.currentDataset = this.dataset[this.datasetLookup[0].id];
+  }
+
+  async getMapData(): Promise<Any[]> {
+    const { http, dataset, files } = this;
+    const lookup: Any = [];
+
+    // Fetch the index.json file
+    const datasetDirectory: any = await http.get('assets/datasets/index.json').toPromise();
+
+    // For each directory in the idnex.json file, go through and fetch the map data.
+    datasetDirectory.forEach((tempDataset: Any) => {
```

Comment:
Why in the world are you loading in all the map data at startup? You should only load a dataset as needed (possibly caching, but maybe not depending on file sizes)